samples/CommunityToolkit.Maui.Sample/App.xaml.cs

```diff
@@ -9,6 +9,7 @@ public App()
 	{
 		InitializeComponent();
 
-		MainPage = new BaseNavigationPage(new MainGalleryPage());
+		var mainGalleryPage = ServiceProvider.GetRequiredService<MainGalleryPage>();
```

Comment:
I think it can be injected into constructor

Reply:
You're not wrong. However, injecting `MainGalleryPage` into `App.cs` creates an order-of-operations problem in .NET MAUI Apps that also use `Application.Resources.ResourceDictionary` in `App.xaml.cs` + XAML Compilation.

The DI container initializes `App` last, and thus its ResourceDictionary also gets initialized last.

This then causes every `ContentPage` that uses `{StaticResource ...}` (e.g. `{StaticResource ContentPadding}`) to throw an error because the `StaticResource` (e.g. `ContentPadding`) doesn't exist during initialization.

Reply:
If we turn off XAML Compilation or stop using `ResourceDictionary` in App.xaml.cs`, then we could inject `MainGalleryPage` into `App`.

Reply:
In the `CommunityToolkit.Maui.Markup` sample app, we **do** inject `MainPage` into the `App` constructor.

This works fine because the app is not using `App.xaml.cs`.

https://github.com/CommunityToolkit/Maui.Markup/blob/ed7ca904ace3b711ea40c45283111b70ac6c606d/samples/CommunityToolkit.Maui.Markup.Sample/App.cs#L10